components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/api/impl/BusinessRulesApiServiceImpl.java

```diff
@@ -0,0 +1,422 @@
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.business.rules.core.api.impl;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.business.rules.core.api.BusinessRulesApiService;
+import org.wso2.carbon.business.rules.core.api.NotFoundException;
+import org.wso2.carbon.business.rules.core.bean.BusinessRule;
+import org.wso2.carbon.business.rules.core.bean.RuleTemplate;
+import org.wso2.carbon.business.rules.core.bean.TemplateGroup;
+import org.wso2.carbon.business.rules.core.bean.TemplateManagerInstance;
+import org.wso2.carbon.business.rules.core.bean.scratch.BusinessRuleFromScratch;
+import org.wso2.carbon.business.rules.core.bean.template.BusinessRuleFromTemplate;
+import org.wso2.carbon.business.rules.core.exceptions.BusinessRuleNotFoundException;
+import org.wso2.carbon.business.rules.core.exceptions.RuleTemplateScriptException;
+import org.wso2.carbon.business.rules.core.exceptions.TemplateManagerServiceException;
+import org.wso2.carbon.business.rules.core.services.TemplateManagerService;
+import org.wso2.carbon.business.rules.core.util.TemplateManagerConstants;
+import org.wso2.carbon.business.rules.core.util.TemplateManagerHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.Response;
+
+/**
+ * Implementation of business rules REST API
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaMSF4JServerCodegen",
+        date = "2017-10-13T06:19:32.032Z")
+public class BusinessRulesApiServiceImpl extends BusinessRulesApiService {
+    private static final Logger log = LoggerFactory.getLogger(BusinessRulesApiServiceImpl.class);
+
+    @Override
+    public Response createBusinessRule(String businessRule, Boolean shouldDeploy
+    ) throws NotFoundException {
+        TemplateManagerService templateManagerService = TemplateManagerInstance.getInstance();
+        // convert the string received from API, as a json object
+        Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+        JsonObject businessRuleJson = gson.fromJson(businessRule, JsonObject.class);
+        int status;
+        List<Object> responseData = new ArrayList<Object>();
```

Comment:
Why `Object`? Seems like you are only putting `String`s into this list.

Reply:
The first two array members are strings denoting error message heading & subheading. The third member is the response, which can be an Object or an Array. That's the purpose of having Object